tests/rest_scripts/abort_multipart_upload.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+# Copyright 2024 Versity Software
+# This file is licensed under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+source ./tests/rest_scripts/rest.sh
+
+# Fields
+
+# shellcheck disable=SC2153
+bucket_name="$BUCKET_NAME"
+# shellcheck disable=SC2153
+key="$OBJECT_KEY"
+# shellcheck disable=SC2153,SC2034
+upload_id="$UPLOAD_ID"
+
+# Step 1:  generate canonical request hash
+
+current_date_time=$(date -u +"%Y%m%dT%H%M%SZ")
+
+canonical_request="DELETE
+/$bucket_name/$key
+uploadId=$UPLOAD_ID
+host:$host
+x-amz-content-sha256:UNSIGNED-PAYLOAD
+x-amz-date:$current_date_time
+
+host;x-amz-content-sha256;x-amz-date
+UNSIGNED-PAYLOAD"
+
+canonical_request_hash="$(echo -n "$canonical_request" | openssl dgst -sha256 | awk '{print $2}')"
+
+# Step 2:  create STS data string
+
+year_month_day="$(echo "$current_date_time" | cut -c1-8)"
+
+sts_data="AWS4-HMAC-SHA256
+$current_date_time
+$year_month_day/$aws_region/s3/aws4_request
+$canonical_request_hash"
+
+# Step 3:  generate signature
+
+date_key=$(echo -n "$year_month_day" | openssl dgst -sha256 -mac HMAC -macopt key:"AWS4${aws_secret_access_key}" | awk '{print $2}')
```

Comment:
This should get wrapped in a function somewhere since the same code is used in another test file.